pkg/api/server/v1alpha2/results_test.go

```diff
@@ -102,6 +104,69 @@ func TestCreateResult(t *testing.T) {
 	}
 }
 
+func TestUpdateResult(t *testing.T) {
+	srv, err := New(test.NewDB(t))
+	if err != nil {
+		t.Fatalf("failed to create server: %v", err)
+	}
+	ctx := context.Background()
+
+	tt := []struct {
+		name    string
+		create  *pb.Result
+		update  *pb.Result
+		expect  *pb.Result
+		errcode codes.Code
+	}{
+		{
+			name:   "test success",
+			update: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+		},
+		{
+			name:   "test update with empty result",
+			expect: &pb.Result{},
+		},
+	}
+	for idx, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// create a result for test.
+			r := &pb.Result{Name: fmt.Sprintf("foo/results/bar-%v", idx)}
```

Comment:
We can probably just inline this below - the returned `created` can be used everywhere else `r` is used.